### PR TITLE
[TensorMerge] Add  draft tensor_merge plugin @open sesame 10/10 10:41

### PR DESCRIPTION
**Pull request by jijoongmoon**
# PR Description
we need tensor_merge to merge tensor. It make multiple tensor into
single tensor. After this PR, C:W:H:B ([B][H][W][C]) + C:W:H:B ([B][H][W][C]) --> W:H:2xC:B ([B][2xC][H][W]).
Therefore, if you want to use this as input for tensorflow lite, you may need transform element,
with transpose mode. ( mode=transpose, option=2:0:1:3. It produce 2xC:W:H:B ([B][H][W][2xC]). 

**Changes proposed in this PR:**
- Added Tensor merge plugin

**TODO**
- Add Multiple direction merge : Width, Height or mixed.
- Better handling to reduce memcopy.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: gst/tensor_merge/gsttensormerge.c [hunk 1]
```diff
@@ -0,0 +1,793 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	gsttensormerge.c
+ * @date	03 July 2018
+ * @brief	GStreamer plugin to merge tensors (as a filter for other general neural network filters)
+ * @see		https://github.com/nnsuite/nnstreamer
+ * @see		https://github.sec.samsung.net/STAR/nnstreamer
```
Comment (myungjoo) [inline gst/tensor_merge/gsttensormerge.c]: Please remove this broken link.

File: gst/tensor_merge/gsttensormerge.c [hunk 2]
```diff
@@ -0,0 +1,793 @@
...
+static tensor_merge_linear_mode
+gst_tensor_merge_get_linear_mode (const gchar * str)
+{
+  int i;
+  for (i = 0; i < LINEAR_END; i++) {
+    if (!g_ascii_strcasecmp (gst_tensor_merge_linear_string[i], str))
+      return i;
+  }
+  return LINEAR_END;
+}
```
Comment (myungjoo) [inline gst/tensor_merge/gsttensormerge.c]: You can reuse ```find_key_strv``` with NULL termination.

File: gst/tensor_merge/gsttensormerge.h [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	gsttensormerge.h
+ * @date	03 July 2018
+ * @brief	GStreamer plugin to merge tensors (as a filter for other general neural network filters)
+ * @see		https://github.com/nnsuite/nnstreamer
+ * @see		https://github.sec.samsung.net/STAR/nnstreamer
```
Comment (myungjoo) [inline gst/tensor_merge/gsttensormerge.h]: Please remove this broken URL

File: gst/tensor_merge/gsttensormerge.c [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+    segment.start = time;
+    gst_pad_push_event (tensor_merge->srcpad, gst_event_new_segment (&segment));
+    tensor_merge->need_segment = FALSE;
+  }
+
+
+  /* TODO : Should handle Properly. This is just for the temperaly usage */
+  mem = gst_buffer_get_memory_range (tensors_buf, 0, -1);
+  gst_buffer_append_memory (tensor_buf, mem);
+
```
Comment (jaeyun-jung) [inline gst/tensor_merge/gsttensormerge.c]: Please copy timestamp from tensors_buf after appending memory block.

gst_buffer_copy_into (tensor_buf, tensors_buf, GST_BUFFER_COPY_TIMESTAMPS, 0, -1);

Comment (jaeyun-jung) [reply gst/tensor_merge/gsttensormerge.c]: Also, add code to release tensors_buf. 

gst_buffer_unref (tensors_buf);

**Comment by jijoongmoon** [comment]
I think it is ready for further review! Thanks.

File: gst/tensor_merge/gsttensormerge.c [hunk 5]
```diff
@@ -0,0 +1,780 @@
...
+          break;
+        case LINEAR_WIDTH:
+        case LINEAR_HEIGHT:
+        default:
+          ret = FALSE;
+      }
+    }
+      break;
+    default:
+      if (!ret)
```
Comment (jaeyun-jung) [inline gst/tensor_merge/gsttensormerge.c]: Please consider this line. 
If ret=GST_FLOW_NOT_SUPPORTED supposed error, but it will run as true.

**Pull request merged**